src/services/tx/tx-sender/dispatch.ts

```diff
@@ -59,11 +59,11 @@ export const dispatchTxProposal = async ({
  */
 export const dispatchTxSigning = async (
   safeTx: SafeTransaction,
-  shouldEthSign: boolean,
+  isLegacySignMethod: boolean,
   txId?: string,
 ): Promise<SafeTransaction> => {
   const sdk = getAndValidateSafeSDK()
-  const signingMethod = shouldEthSign ? 'eth_sign' : 'eth_signTypedData'
+  const signingMethod = isLegacySignMethod ? ('eth_signTypedData_v3' as 'eth_signTypedData') : 'eth_signTypedData'
```

Comment:
does it work for other hardware wallets?

Reply:
the way it worked in `safe-react` is that it tried `eth_signTypedData_v4`, then `v3`, then `eth_sign`. If you try an unsupported method it usually throws `METHOD_NOT_SUPPORTED` error

Reply:
Yeah, we don't have that cascading try-catch now, the if was catching the cases we knew about.
Richard says we shouldn't use v3 because it's not fully implementing the EIP-712.